packages/firestore/lite/src/api/reference.ts

```diff
@@ -188,6 +304,10 @@ export function collection(
   validateArgType('doc', 'non-empty string', 2, relativePath);
   const path = ResourcePath.fromString(relativePath);
   if (parent instanceof Firestore) {
+    // Kick off configuring the client, which freezes the settings.
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+    parent._ensureClientConfigured();
```

Comment:
This matches the main SDK (we freeze the settings in Firestore.doc) and is needed to be able to call "newUserDataReader" in the Query constructor.

Reply:
This can actually be part of `newUserDataReader`. I also update the Settings and Datastore APIs to be more type-system strict (e.g. remove the bang on `settings`). To do this, I split `ensureClientConfigured` into `getSettings` and `getDatastore`.